cvmfs/catalog_mgr_rw.cc

```diff
@@ -172,29 +172,34 @@ manifest::Manifest *WritableCatalogManager::CreateRepository(
 
 
 /**
- * Tries to retrieve the catalog containing the given path
- * This method is just a wrapper around the FindCatalog method of
- * AbstractCatalogManager to provide a direct interface returning
- * WritableCatalog classes.
- * @param path the path to look for
- * @param result the retrieved catalog (as a pointer)
- * @return true if catalog was found, false otherwise
+ * Retrieve the catalog containing the given path.
+ * Other than AbstractCatalogManager::FindCatalog() this mounts nested
+ * catalogs if necessary and returns  WritableCatalog objects.
+ * Furthermore it optionally returns the looked-up DirectoryEntry.
+ *
+ * @param path    the path to look for
+ * @param result  the retrieved catalog (as a pointer)
+ * @param dirent  is set to looked up DirectoryEntry for 'path' if non-NULL
+ * @return        true if catalog was found
  */
-bool WritableCatalogManager::FindCatalog(const string &path,
-                                         WritableCatalog **result)
-{
+bool WritableCatalogManager::FindCatalog(const string     &path,
+                                         WritableCatalog **result,
+                                         DirectoryEntry   *dirent) {
```

Comment:
This method retrieves the `Catalog` containing `path` and then does a lookup to ensure that `path` is available. Sometimes the `DirectoryEntry` of this lookup is needed in the caller as well. The (optional) `dirent` out-parameter allows the caller to obtain it directly and safe an expensive call to `Lookup()`.